src/mlpack/methods/bayesian_linear_regression/bayesian_linear_regression_main.cpp

```diff
@@ -0,0 +1,191 @@
+    "Parameters are tuned during the maximization of the marginal likelihood. "
+    "This procedure includes the Ockham's razor that penalizes over complex "
+    "solutions. "
+    "\n\n"
+    "This program is able to train a Bayesian linear regression model or load "
+    "a model from file, output regression predictions for a test set, and save "
+    "the trained model to a file."
+    "\n\n"
+    "To train a BayesianLinearRegression model, the " +
+    PRINT_PARAM_STRING("input") + " and " + PRINT_PARAM_STRING("responses") +
+    "parameters must be given. The " + PRINT_PARAM_STRING("center") +
+    "and " + PRINT_PARAM_STRING("scale") + " parameters control the "
+    "centering and the normalizing options. A trained model can be saved with "
+    "the " + PRINT_PARAM_STRING("output_model") + ". If no training is desired "
+    "at all, a model can be passed via the " +
+    PRINT_PARAM_STRING("input_model") + " parameter."
+    "\n\n"
+    "The program can also provide predictions for test data using either the "
+    "trained model or the given input model.  Test points can be specified "
+    "with the " + PRINT_PARAM_STRING("test") + " parameter.  Predicted "
+    "responses to the test points can be saved with the " +
+    PRINT_PARAM_STRING("predictions") + " output parameter. The "
+    "corresponding standard deviation can be save by precising the " +
+    PRINT_PARAM_STRING("stds") + " parameter."
+    "\n\n"
+    "For example, the following command trains a model on the data " +
+    PRINT_DATASET("data") + " and responses " + PRINT_DATASET("responses") +
+    "with center set to true and scale set to false (so, Bayesian "
+    "linear regression is being solved, and then the model is saved to " +
+    PRINT_MODEL("bayesian_linear_regression_model") + ":"
+    "\n\n" +
+    PRINT_CALL("bayesian_linear_regression", "input", "data", "responses",
+               "responses", "center", 1, "scale", 0, "output_model",
+               "bayesian_linear_regression_model") +
+    "\n\n"
+    "The following command uses the " +
+    PRINT_MODEL("bayesian_linear_regression_model") + " to provide predicted " +
+    " responses for the data " + PRINT_DATASET("test") + " and save those " +
+    " responses to " + PRINT_DATASET("test_predictions") + ": "
+    "\n\n" +
+    PRINT_CALL("bayesian_linear_regression", "input_model",
+               "bayesian_linear_regression_model", "test", "test",
+               "predictions", "test_predictions") +
+    "\n\n"
+    "Because the estimator computes a predictive distribution instead of "
+    "simple point estimate, the " + PRINT_PARAM_STRING("stds") + " parameter "
+    "allows to save the prediction uncertainties: "
+    "\n\n" +
+    PRINT_CALL("bayesian_linear_regression", "input_model",
+               "bayesian_linear_regression_model", "test", "test",
+               "predictions", "test_predictions", "stds", "stds"));
```

Comment:
Hi @rcurtin, For consistency point of view, should we add some `SEE_ALSO()` here?